src/Parser/DemonHunter/Havoc/Modules/Items/DelusionsOfGrandeur.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import SpellLink from 'common/SpellLink';
+import { formatNumber, formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+import Wrapper from 'common/Wrapper';
+import FuryTracker from '../ResourceTracker/FuryTracker';
+
+/*
+* Equip: The remaining cooldown on Metamorphosis is reduced by 1 sec for every 30 Fury you spend.
+*/
+
+class DelusionsOfGrandeur extends Analyzer {
+	static dependencies = {
+		combatants: Combatants,
+		SpellUsable: SpellUsable,
+		furyTracker: FuryTracker,
+	}
+
+	on_initialized() {
+		this.active = this.combatants.selected.hasShoulder(ITEMS.DELUSIONS_OF_GRANDEUR.id);
+	}
+
+	get cooldownReductionRatio(){
+		return this.furyTracker.cooldownReduction / (this.owner.fightDuration + this.furyTracker.cooldownReduction);
```

Comment:
Uhh, I don't think that's what he meant by %CDR. I'll use concrete examples to explain the suggestion I made and the one emallson made:

**Percent CDR**
Meta's CD is 5 minutes, but with DoG you proc enough cooldown reduction that it comes off CD after 4 minutes. `(5 - 4) / 5 = 0.20` ... You display something like "This reduced Meta's CD by 20%". If there were multiple casts you would average the CDR.

**Cooldown Reduced per Cast**
Meta's CD is 5 minutes. After your first cast, DoG reduces the cooldown enough that it comes up after 4 minutes (so reduced CD by 60 seconds). After the second, it comes up after 3:30 (so reduced by 90 seconds). Average reduction was 75 seconds. You display something like "This reduced Meta's CD by an average of 75 seconds".

Reply:
I can do that. Is I included abilities.js can I the cooldown of meta? Or is there a better way to do that since it's affected by traits

Reply:
You can just build it in the on_initialized

Reply:
The way he has it is actually how I do it in the brewmaster code. Semantically, the numerator is the amount of time gained, the denominator is the total "time" that passed for this abilities cooldown. Note that:

1. It is 0% when no CDR was applied
2. It is 50% if time gained = fight time
3. It is 100% only if infinity time was gained (which can't happen, lining up with the fact that we can't get 100% CDR with "reduces CD by X seconds" tied to separate abilities)

This also lines up with what you wrote: if you gained 1 minute of reduction over 4 minutes of fight time, then `1 / (4 + 1) = 0.2`. If the total fight length was 6 minutes, you would gain an additional 30 seconds of time *if the CDR gain were perfectly continuous* at that rate, giving a CDR of `1.5 / (6 + 1.5) = 0.2`. The way the item is worded, the continuity assumption looks not unreasonable.

**However** my use case is for abilities with very short (< 20s) cooldowns. After thinking about it, this formula may not make the most sense for Meta because even if you do reduce the CD by 1.5 minutes total, you still may not get an extra cast. Not sure how to deal with that problem.